packages/qwik/src/core/use/use-resource.ts

```diff
@@ -191,8 +191,8 @@ export const useResource$ = <T>(
 /** @public */
 export interface ResourceProps<T> {
   readonly value: ResourceReturn<T> | Signal<Promise<T> | T> | Promise<T>;
-  onResolved: (value: T) => JSXOutput;
-  onPending?: () => JSXOutput;
+  onResolved: (value: T) => JSXOutput | Promise<JSXOutput>;
+  onPending?: () => JSXOutput | Promise<JSXOutput>;
```

Comment:
I don't understand why this is needed. The resource holds a Promise, but the Resource component should just render without a Promise

Reply:
From what I know, it is correct for JavaScript, but it is not correct for TypeScript.
TypeScript will think that the function returns a Promise when async is added before the function.

Reply:
@wmertens

Reply:
you shouldn't add async before the function then

Reply:
aaah ok now I get it, it's when the dev puts an async in the resource functions, ok. Hmm not sure if that's a good idea though - the render functions are just like inline components and we don't really support async inline components either. It works though :)

@varixo thoughts?